lib/matplotlib/cbook.py

```diff
@@ -2661,3 +2661,170 @@ def __exit__(self, exc_type, exc_value, traceback):
+
+    def __init__(self, str_func):
+        """
+        Parameters
+        ----------
+        str_func : string
+            String to be parsed.
+
+        """
+        try:  # For python 2.7 and python 3+ compatibility
+            is_str = isinstance(str_func, basestring)
+        except NameError:
+            is_str = isinstance(str_func, str)
+
+        if not is_str:
+            raise ValueError("The argument passed is not a string.")
+        self._str_func = str_func
+        self._key, self._params = self._get_key_params()
+        self._func = self.get_func()
+
+    def get_func(self):
+        """
+        Returns the _FuncInfo object, replacing the relevant parameters if
+        necessary in the lambda functions.
+
+        """
+
+        func = self._funcs[self._key].copy()
+        if len(self._params) > 0:
+            m = func.direct
+            func.direct = (lambda x, m=m: m(x, self._params))
+            m = func.inverse
+            func.inverse = (lambda x, m=m: m(x, self._params))
+        return func
+
+    def get_directfunc(self):
+        """
+        Returns the callable for the direct function.
+
+        """
+        return self._func.direct
+
+    def get_invfunc(self):
+        """
+        Returns the callable for the inverse function.
+
+        """
+        return self._func.inverse
+
+    def is_bounded_0_1(self):
+        """
+        Returns a boolean indicating if the function is bounded
+        in the [0-1 interval].
+
+        """
+        return self._func.bounded_0_1
+
+    def _get_key_params(self):
+        str_func = six.text_type(self._str_func)
+        # Checking if it comes with parameters
+        regex = '\{(.*?)\}'
+        params = re.findall(regex, str_func)
+
+        if len(params) > 0:
+            for i in range(len(params)):
+                try:
+                    params[i] = float(params[i])
+                except:
+                    raise ValueError("'p' in parametric function strings must"
+                                     " be replaced by a number that is not "
+                                     "zero, e.g. 'log10(x+{0.1})'.")
+
+                if params[i] == 0:
+                    raise ValueError("'p' in parametric function strings must"
+                                     " be replaced by a number that is not "
+                                     "zero.")
+            str_func = re.sub(regex, '{p}', str_func)
+
+        try:
+            func = self._funcs[str_func]
+        except KeyError:
+            raise ValueError("%s: invalid function. The only strings "
+                             "recognized as functions are %s." %
+                             (str_func, self.funcs.keys()))
+        except:
+            raise ValueError("Invalid function. The only strings recognized "
+                             "as functions are %s." %
+                             (self.funcs.keys()))
+        if len(params) > 0:
+            func.direct(0.5, params)
```

Comment:
Sure, I actually though about this more and there is no way I could fail there, because the implementation guarantees that the parameters are floating point numbers. And the fact that there  is a mathing string guarantees that the number of parameters is right.

What I am going to include is the possibility of checking if the parameters are in the right range, and also checking wheter the fucntion is bounded as fucntion of the parameters. This way, we can also allow parameters equal to 0.